### PR TITLE
add unsorted option

### PR DESCRIPTION
**Pull request by PeterTheOne**
As seen by @HelmutLeitner in https://github.com/HelmutLeitner/SemanticCompoundQueries/commit/4a8b0cec153e7dda1d96e5c8abf4dd3311072375.

Old PR: https://github.com/wikimedia/mediawiki-extensions-SemanticCompoundQueries/pull/2

### CODE REVIEW

File: SCQ_QueryProcessor.php [hunk 1]
```diff
@@ -76,7 +76,10 @@ public static function doCompoundQuery( Parser &$parser ) {
 		}
 
 		// Sort results so that they'll show up by page name
-		uasort( $results, array( 'SCQQueryProcessor', 'compareQueryResults' ) );
+		$unsorted = $other_params['unsorted'];
```
Comment (JeroenDeDauw) [inline SCQ_QueryProcessor.php]: No need to introduce a local var here

**Comment by PeterTheOne** [comment]
Is there a standard way of checking if a variable is `yes`, `true` or `on` in mediawiki or semanticmediawiki?

**Comment by PeterTheOne** [comment]
Ok, there seem to be `smw_true_words` and `smw_false_words` in i18n and https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/src/DataValues/BooleanValue.php uses them.

**Comment by JeroenDeDauw** [comment]
It'd be nice to use that I guess, though SMW does not provide a sane interface to do so. Using a SMW system message that you than throw through MediaWiki's i18n stuff is a very different responsibility (and one with huge dependencies) than where you'd need it here (and presumably elsewhere). If you want to use it here, you want an interface like

```php
interface BooleanInputParser {
    public function isTrue( string $input ): bool;
}
```

In SMW, which then also implements it and provides access to the instance via some factory/registry, that in turn is then preferably made accessible via non-global scope to the code here somehow.

Ping @mwjames in case you are interested

**Comment by PeterTheOne** [comment]
> is a very different responsibility (and one with huge dependencies) than where you'd need it here

Agreed. Would be nice to have something like that, how does Maps do this, through `DataValues/BooleanValue`?

**Comment by JeroenDeDauw** [comment]
> how does Maps do this, through DataValues/BooleanValue?

Yeah - it's handled via ParamProcessor. No i18n support there tho.

**Pull request merged**